django/santropolFeast/member/views.py

```diff
@@ -315,6 +319,96 @@ def get_context_data(self, **kwargs):
 
         return context
 
+    def get(self, request, **kwargs):
+
+        self.format = request.GET.get('format', False)
+
+        if self.format == 'csv':
+            return ExportCSV(
+                self, self.get_queryset()
+            )
+
+        return super(ClientList, self).get(request, **kwargs)
+
+
+def ExportCSV(self, queryset):
+    reponse = HttpResponse(content_type="text/csv")
+    reponse['Content-Disposition'] =\
+        'attachment; filename=client_export.csv'
+    writer = csv.writer(reponse, csv.excel)
+
+    writer.writerow([
+        "ID",
+        "Client Firstname",
+        "Client Lastname",
+        "Client Status",
+        "Client Alert",
+        "Client Gender",
+        "Client Birthdate",
+        "Client Delivery",
+        "Client Home Phone",
+        "Client Cell Phone",
+        "Client Work Phone",
+        "Client Email",
+        "Client Street",
+        "Client Apartment",
+        "Client City",
+        "Client Postal Code",
+        "Client Route",
+        "Client Billing Type",
+        "Billing Firstname",
+        "Billing Lastname",
+        "Billing Street",
+        "Billing Apartment",
+        "Billing City",
+        "Billing Postal Code",
+        "Emergency Contact Firstname",
+        "Emergency Contact Lastname",
+        "Emergency Contact Home Phone",
+        "Emergency Contact Cell Phone",
+        "Emergency Contact Work Phone",
+        "Emergency Contact Email",
+        "Emergency Contact Relationship",
+        "Meal Default",
+    ])
+
+    for obj in queryset:
+        writer.writerow([
+            obj.id,
+            obj.member.firstname,
+            obj.member.lastname,
+            obj.status,
+            obj.alert,
+            obj.gender,
+            obj.birthdate,
+            obj.delivery_type,
+            obj.member.home_phone,
+            obj.member.cell_phone,
+            obj.member.work_phone,
+            obj.member.email,
+            obj.member.address.street,
+            obj.member.address.apartment,
+            obj.member.address.city,
+            obj.member.address.postal_code,
+            obj.route.name,
```

Comment:
If no route is set, the export crashes. Could you find a way to prevent this to happen ? A client may have no route assigned.